### PR TITLE
Add support for zwave-js 10

### PR DESCRIPTION
**Pull request by raman325**
- Remove old controller attributes and replaced with new ones
- Switch controller.begin_exclusion to use exclusion strategy instead of unprovision flag
- Add API key support to `controller.get_available_firmware_updates`
- Add `node.interview` command
- Update `NodeFoundEventModel` to use new TypedDict

### CODE REVIEW

**Comment by raman325** [comment]
coverage is related to a `None` check

**Review by MartinHjelmare — COMMENTED** [review]
I'm missing these changes from the migration guide:

- https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v10?id=changed-argument-type-of-quotnode-foundquot-event

File: zwave_js_server/model/node/data_model.py [hunk 1]
```diff
@@ -14,6 +14,13 @@
     from typing import TypedDict
 
 
+class FoundNodeDataType(TypedDict, total=False):
+    """Represent a found node data dict type."""
+
+    nodeId: int
+    deviceClass: DeviceClassDataType
```
Comment (MartinHjelmare) [inline zwave_js_server/model/node/data_model.py:21]: What about supported CCs and controlled CCs?

https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating-to-v10?id=changed-argument-type-of-quotnode-foundquot-event

**Review by MartinHjelmare — APPROVED** [review]
Looks good!

File: zwave_js_server/model/node/data_model.py (hunk 1 above)
Comment (MartinHjelmare) [reply zwave_js_server/model/node/data_model.py:21]: We don't currently include these in the server when building the node found event data.

Comment (raman325) [reply zwave_js_server/model/node/data_model.py:21]: added a todo for that https://github.com/zwave-js/zwave-js-server/issues/722

**Pull request merged**